src/herbpy/barretthand.py

```diff
@@ -125,17 +125,27 @@ def MoveHand(self, f1=None, f2=None, f3=None, spread=None, timeout=None):
         @param spread spread angle, in radians
         @param timeout blocking execution timeout, in seconds
         """
-        # Default any None's to the current DOF values.
-        preshape = self.GetDOFValues()
-        if f1     is not None: preshape[0] = f1
-        if f2     is not None: preshape[1] = f2
-        if f3     is not None: preshape[2] = f3
-        if spread is not None: preshape[3] = spread
+        curr_pos = self.GetDOFValues()
+        preshape = [None]*4
+        # Set control command and
+        # default any None's to the current DOF values.
+        if self.simulated:
+            preshape[0] = spread if spread is not None else curr_pos[0]
+            preshape[1] = f1 if f1 is not None else curr_pos[1]
+            preshape[2] = f2 if f2 is not None else curr_pos[2]
+            preshape[3] = f3 if f3 is not None else curr_pos[3]
+        else:
+            # TODO when PositionCommandController is updated to index by
+            # joint name, this special logic can be removed
+            preshape[0] = f1 if f1 is not None else curr_pos[1]
+            preshape[1] = f2 if f2 is not None else curr_pos[2]
+            preshape[2] = f3 if f3 is not None else curr_pos[3]
+            preshape[3] = spread if spread is not None else curr_pos[0]
 
         self.controller.SetDesired(preshape)
         util.WaitForControllers([ self.controller ], timeout=timeout)
 
-    def OpenHand(hand, spread=None, timeout=None):
+    def OpenHand(hand, spread=0.0, timeout=None):
```

Comment:
The purpose of this function is to open the hand without changing the spread (`None` specifies "don't move this joint") unless a spread value is explicitly passed as an argument. This change causes `OpenHand` to *always* move the spread to zero.